### ISSUE TITLE
Mac Build

### ISSUE THREAD

**Issue by maikramer** [maikramer/Materialize#6]
## Mac Build
 I will post here some fast builds to mac, that i can't test because the lack of a mac computer.
The builds will come from the development branch, based on HDRP, just because i'm dont want to work in 2 branches at the same time.

### Caution
- In mac HDRP uses metal, while in the Windows and Linux platform they are based on Vulkan.
- At least on windows, using directx instead of vulkan took the program to generate wrong colors, then i changed to vulkan on windows, but i dont have this option on mac.

### Won't be adressed soon
- Paste directly from clipboard only works on linux and i didn't got it to work on windows for now, and windows i can test in my own computer.

**Comment by fouman75** [comment]
@maikramer @BenRacicot I was curious and I built v0.3.7 on my mac. Everything seems to work fine as far as I can tell on OSX 10.14.5.

I put the compiled version in [dropbox](https://www.dropbox.com/s/4buvficjn2tfxob/materialize_0.3.7.app.zip?dl=0). 

Hope this helps!

**Comment by maikramer** [comment]
Thanks people, i'm going home tomorrow(I work in a oil company), i will try to build in my linux computer. But, i see that @fouman75 already did a great job. I will put his file in the release page, then can be easier to find. Tomorow, i cant use dropbox here. Thenks fourman, now the multiplatform is complete. I wont implement new things but i will keep the build updated.

**Comment by jeffg2k** [comment]
I noticed that [MoltenVK](https://github.com/KhronosGroup/MoltenVK/releases/) is open source, which supposedly adds Vulcan support to Mac.  If it's difficult to maintain or build in the support for Metal, I wonder if a MoltenVK prerequisite would work.  I'm very interested in a mac version. 👍

**Comment by fouman75** [comment]
@jeffg2k 

Please see https://github.com/maikramer/Materialize/issues/10

There is already a build for OSX here: [Dropbox](https://www.dropbox.com/s/4buvficjn2tfxob/materialize_0.3.7.app.zip?dl=0)

It should work but you might need to turn off post processing as mentioned in issue 10 if it's not turned off by default.

**Comment by jeffg2k** [comment]
@fouman75 thanks, I gave it a try.  The resolution on the retina display was too small to work with, probably trying to do 2880x1800.  It also launched in full screen and the windowed option was grayed out. I wasn't able to shrink it down or adjust the resolution to normal (or double density) to make it easily usable.

**Comment by fouman75** [comment]
@jeffg2k Did you try using "cmd-f" to get out of full screen and then resize the window? The app does use the default resolution of your computer.

**Comment by jeffg2k** [comment]
Yes, I was able to get out of fullscreen, but yeah - I'm on a 15" macbook pro and the native resolution for retina is 2880x1800, which is too high for 15" if the programs are not aware to rescale to 1440x900 or double up on their DPI.

**Comment by fouman75** [comment]
@jeffg2k Can you try starting the app from the command line?
./Materialize_0.3.7.app/Contents/MacOS/Materialize -screen-height 900 -screen-width 1440

You can change the resolution through arguments. Let me know if it works out on your mac. I might add a resolution dialog if you confirm that it's usable with a resolution change.

**Comment by fouman75** [comment]
The problem is that the UI has been created manually with immediate mode GUI and not Unity's Canvas UI which supports resolution scaling. I feel like the UI must be redone from scratch and it's not something I want to get into at this point.

**Comment by maikramer** [comment]
Yes. I started exactly doing this, migrating to Unity UI, but I didn't
finish. I migrated all shaders to compute shaders also, but didn't finish
either, if somebody wants to take a look at the HDRP branch, there is a lot
of experiments there, but, I have a lot of projects as electronic engineer,
and I have my job as Automation technician, just too many things for me.
I'm sorry about the project guys...

Em seg, 24 de ago de 2020 16:11, Frederick Ouimet <notifications@github.com>
escreveu:

**Comment by fouman75** [comment]
No worries @maikramer, you did some great improvements already. I'm curious to how know many people actually use this software. With all the free and paid PBR materials available and software like SubstancePainter, I feel that there is little use for Materialize nowadays. It would still be interesting to know how people integrate this in their workflow. Maybe it would be worth it to improve Materialize if it's really useful to a lot of people.

**Comment by maikramer** [comment]
Yes, I think this also, but I think Linux users like me don't have so many
options like Mac and Windows users. There is no Substance or something like
that in Linux. Anyway, for Linux users, the software is feature complete I
think. Mac users have some problems, but, implement a UI scaling won't be
too difficult even using IM Gui.

Em ter, 25 de ago de 2020 12:10, Frederick Ouimet <notifications@github.com>
escreveu:
